src/Illuminate/Database/Eloquent/Relations/MorphTo.php

```diff
@@ -181,7 +181,7 @@ protected function gatherKeysByType($type, $keyType)
      */
     public function createModelByType($type)
     {
-        $class = Model::getActualClassNameForMorph($type);
+        $class = $this->parent::getActualClassNameForMorph($type);
```

Comment:
If the subclass does not define `getActualClassNameForMorph()`, it will still call `Model::getActualClassNameForMorph()`, Doing this allows subclasses to override `getActualClassNameForMorph()`, avoiding the dependency on `Relation::$morphMap`